p2p/client.go

```diff
@@ -421,3 +503,75 @@ func (c *Client) bootstrapLoop(ctx context.Context) {
 		}
 	}
 }
+
+// latest seen height to be retrieved using block-sync
+func (c *Client) SetLatestSeenHeight(height uint64) {
+	if height > c.latestSeenHeight {
+		c.latestSeenHeight = height
+	}
+}
+
+// already applied height set as a min height to retrieved using block-sync
+func (c *Client) SetAppliedHeight(height uint64) {
+	defer c.blockAppliedHeightMu.Unlock()
+	c.blockAppliedHeightMu.Lock()
+
+	if height > c.appliedHeight {
+		for h := height; h <= c.appliedHeight; h++ {
+			delete(c.heightToSkip, h)
+		}
+		c.appliedHeight = height
+	}
+}
+
+// Loop used to request missing blocks via block-sync
+func (c *Client) BlockSyncLoop(ctx context.Context) {
```

Comment:
IMO this function should **query** data on demand from the BlockManager
(e.g `c.node.GetBlockManagerHeight()` used by RPC)
otherwise adding the complexity of calling `SetAppliedHeight` for each block, while even not needed for the happy flow (as block sync is a fallback method after node is running and synced )

Reply:
replaced by query to state in the store